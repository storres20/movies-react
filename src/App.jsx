import { MoviesGrid } from "./MoviesGrid";
import styles from "./App.module.css";

export function App() {
  return <div>
    <header>
      <h1 className={styles.title}>Movies</h1>
    </header>
    <main>
      <MoviesGrid/>
    </main>
  </div>;
}

