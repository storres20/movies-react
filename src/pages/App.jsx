import { MoviesGrid } from "../components/MoviesGrid";
import styles from "./App.module.css";
import {
  BrowserRouter,
  Routes,
  Route,
  Link
} from "react-router-dom";
import AboutPage from "../components/AboutPage";
import NotFoundPage from "../components/NotFoundPage";
import Navbar from "../components/Navbar";
import MovieDetails from "./MovieDetails";

export function App() {
  return (
    <BrowserRouter>
      <Navbar/>
      <header>
        <Link to="/">
          <h1 className={styles.title}>Movies</h1>
        </Link>
      </header>
      <main>
        <Routes>
          <Route path="/" element={<MoviesGrid />} />
          <Route path="/about" element={<AboutPage/>} />
          <Route path="/movies/:movieId" element={<MovieDetails/>} />
          <Route path="/*" element={<NotFoundPage/>} />
        </Routes>
      </main>
    </BrowserRouter>
  );
}



