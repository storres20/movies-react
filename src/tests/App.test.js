import React from 'react';
import renderer from 'react-test-renderer';
import App from '../pages/App';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <App />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
