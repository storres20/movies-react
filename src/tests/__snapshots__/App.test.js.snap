// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-dark"
  >
    <div
      className="container"
    >
      <span
        className="NavTitle navbar-brand"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Movies
        </a>
      </span>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="me-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <a
              aria-current="page"
              className="NavSubTitle active"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <a
              className="NavSubTitle"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </a>
          <div
            className="nav-item dropdown"
          >
            <a
              aria-expanded={false}
              className="dropdown-toggle nav-link"
              data-rr-ui-event-key={null}
              href="#"
              id="basic-nav-dropdown"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              DropdownXXX
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>,
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="title"
      >
        Movies
      </h1>
    </a>
  </header>,
  <main>
    <div>
      <div
        className="divFlex"
      >
        <input
          className="inputFlex"
          onChange={[Function]}
          placeholder="Search..."
          type="text"
        />
      </div>
      <ul
        className="moviesGrid"
      />
    </div>
  </main>,
]
`;
